src/number.h

```diff
@@ -23,6 +30,15 @@ class Number : public Basic {
     virtual bool is_negative() const = 0;
 
     virtual bool is_positive() const = 0;
+
+    virtual bool is_exact() const { return true; };
+    virtual eval_function get_eval_function() const {
+        throw std::runtime_error("Not Implemented.");
+    };
+    virtual Evaluate* get_eval() const {
+        return nullptr;
+    }
```

Comment:
You should never return a raw pointer like this. In this particular case, since Evaluate is a small class with no members, you can just return `Evaluate` by value. In more complicated cases, you need to use a smart pointer, perhaps `std::unique_ptr`.